kernel_gateway/services/kernels/pool.py

```diff
@@ -1,13 +1,20 @@
 # Copyright (c) Jupyter Development Team.
 # Distributed under the terms of the Modified BSD License.
+"""Kernel pools that track and delegate to kernels."""
 
 from tornado.locks import Semaphore
 from tornado import gen
 
 class KernelPool(object):
-    '''
-    Spawns a pool of kernels and cleans them up when shut down.
-    '''
+    """Spawns a pool of kernels and cleans them up when shut down.
```

Comment:
Spawns a pool of kernels. 

The kernel pool is responsible for clean up and shutdown of individual kernels that are members of the pool.